crates/vm/levm/src/gas_cost.rs

```diff
@@ -191,11 +191,15 @@ fn copy_behavior(
         .saturating_sub(1))
         / WORD_SIZE;
 
-    let memory_expansion_cost = current_call_frame.memory.expansion_cost(
-        offset
-            .checked_add(size)
-            .ok_or(OutOfGasError::GasCostOverflow)?,
-    )?;
+    let memory_expansion_cost = if size != 0 {
+        current_call_frame.memory.expansion_cost(
+            offset
+                .checked_add(size)
+                .ok_or(OutOfGasError::GasCostOverflow)?,
+        )?
+    } else {
+        U256::zero()
+    };
```

Comment:
Is this only a copy behavior, or should every memory expansion cost be 0 if the size is 0? If so, we should move this to `Memory::expansion_cost`.

Reply:
The problem here is that `Memory::expansion_cost` takes only one argument. Maybe we can refactor so that it takes something like `(offset, size)` but we have to be careful because I am not sure if everytime we use that in our code we send an offset with size.

Reply:
Yes, I saw that you made some of those changes in your PR about cache memory, and maybe this could be added there. Currently if we had to make that change we would have to change the function signature.

Reply:
You both are right. If it is the right thing to do, proceed with the necessary changes.

Reply:
Done in the the last three PRs (excluding the merge one)